core/src/main/java/org/infinispan/configuration/cache/BackupConfigurationBuilder.java

```diff
@@ -148,6 +148,9 @@ public void validate() {
          throw new CacheConfigurationException("It is required to specify a 'failurePolicyClass' when using a " +
                                                 "custom backup failure policy!");
       }
+      if (useTwoPhaseCommit && strategy == BackupConfiguration.BackupStrategy.ASYNC) {
+         throw new CacheConfigurationException("'two-phase-commit' can be used only with SYNC backup strategy!");
```

Comment:
Ok, let's do it then. @mcimbora, could you update the config javadoc/xsd and i18n the messages?